### PR TITLE
Add environment checks for target machines

### PR DESCRIPTION
**Pull request by yikeke**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/3489
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

File: tiup/tiup-cluster.md [hunk 1]
```diff
@@ -567,3 +567,41 @@ For example, if you previously view the store by running `pd-ctl -u http://127.0
...
+tiup cluster check <cluster-name> --cluster
+```
+
+The CPU thread count check, memory size check, and disk performance check are disabled by default. For the production environment, it is recommended that you enable the three checks and make sure they pass to obtain the best performance.
+
+- CPU: If the number of threads is greater than or equal to 16, the check is passed.
+- Memory: If the total size of physical memory is greater than or equal to 32 GB, the check is passed.
+- Disk: Execute `fio` test on the partitions of `data_dir` and record the results.
+
+When running the checks, if the `--apply` flag is specified, the program automatically tries to repair the failed items. Automatic repair is limited to some items that can be adjusted by modifying the configuration or system parameters. Other unrepaired items need to be handled manually according to the actual situation.
```
Comment (ireneontheway) [inline tiup/tiup-cluster.md:605]: When running the checks, if the `--apply` flag is specified, the program automatically repairs the failed items. Automatic repair is limited to some items that can be adjusted by modifying the configuration or system parameters. Other unrepaired items need to be handled manually according to the actual situation.

File: tiup/tiup-cluster.md [hunk 2]
```diff
@@ -567,3 +567,41 @@ For example, if you previously view the store by running `pd-ctl -u http://127.0
...
+
+Environment checks is not a necessary process for deploying a cluster. For the production environment, it is recommended to perform environment checks and pass all check items before deployment. If not all the check items are passed, the cluster might be deployed and run normally, but the best performance might not be obtained.
```
Comment (ireneontheway) [inline tiup/tiup-cluster.md:607]: An environment check is not necessary for deploying a cluster. For the production environment, it is recommended to perform environment checks and pass all check items before deployment. If not all the check items are passed, the cluster might be deployed and run normally, but the best performance might not be obtained.

File: tiup/tiup-cluster.md (hunk 1 above)
Comment (ireneontheway) [reply tiup/tiup-cluster.md:605]: tries to repair the failed items. >  repairs the failed items.

File: tiup/tiup-cluster.md (hunk 2 above)
Comment (ireneontheway) [reply tiup/tiup-cluster.md:607]: Environment checks is not a necessary process > An environment check

File: tiup/tiup-cluster.md (hunk 1 above)
Comment (yikeke) [inline tiup/tiup-cluster.md:605]:
```suggestion
When running the checks, if the `--apply` flag is specified, the program automatically repairs the failed items. Automatic repair is limited to some items that can be adjusted by modifying the configuration or system parameters. Other unrepaired items need to be handled manually according to the actual situation.
```

File: tiup/tiup-cluster.md (hunk 2 above)
Comment (yikeke) [inline tiup/tiup-cluster.md:607]:
```suggestion
An environment check is not necessary for deploying a cluster. For the production environment, it is recommended to perform environment checks and pass all check items before deployment. If not all the check items are passed, the cluster might be deployed and run normally, but the best performance might not be obtained.
```

**Pull request merged**